caddyhttp/proxy/upstream.go

```diff
@@ -327,11 +327,16 @@ func parseBlock(c *caddyfile.Dispenser, u *staticUpstream) error {
 		if !c.NextArg() {
 			return c.ArgErr()
 		}
-		n, err := strconv.Atoi(c.Val())
+		port := c.Val()
+		n, err := strconv.Atoi(port)
 		if err != nil {
 			return err
 		}
-		u.HealthCheck.Port = n
+
+		if n < 0 {
+			return c.Errf("invalid health_check_port '%s'", port)
+		}
+		u.HealthCheck.Port = c.Val()
```

Comment:
Indeed, it would be nice to be consistent, although it isn't technically wrong... 🤷‍♂️